### PR TITLE
menu dnd event

### PR DESCRIPTION
**Pull request by cspiel1**
This adds an enum `bevent_class` field to `struct bevent` because it should not be derived from the `enum ua_event`. This does not always give wanted results. The second commit gives an example how the fixed API can be used.

- **bevent: add enum bevent_class to struct**
- **menu: emit bevent for dnd reject**

### CODE REVIEW

**Comment by maximilianfridrich** [comment]
Looks good to me.

File: src/bevent.c [hunk 1]
```diff
@@ -531,6 +482,20 @@ int event_encode_dict(struct odict *od, struct ua *ua, enum ua_event ev,
 }
 
 
+static int odict_pl_add(struct odict *od, const char *key,
+			const struct pl *val)
```
Comment (cspiel1) [inline src/bevent.c:486]: Should we add this to `re`?

Comment (cspiel1) [reply src/bevent.c:486]: https://github.com/baresip/re/pull/1208

**Comment by juha-h** [comment]
Does this PR mean that API has changed and existing apps will need to be modified?  If so, the PR should document the steps needed.

**Comment by cspiel1** [comment]
No, the API didn't change.

But now it allows to use the "emit" functions in a more flexible way. E.g. in main only this ev is allowed in first parameter:

```c
bevent_sip_msg_emit(UA_EVENT_SIPSESS_CONN, msg, ...);
```

Now this is also possible:
```c
bevent_sip_msg_emit(UA_EVENT_MODULE, msg, ...);
```

**Comment by juha-h** [comment]
@cspiel1 Yes, but before this commit, there was no `bevent_sip_msg_emit` call in `menu.c`.  Why is it now needed or is it there just for fun?

**Comment by cspiel1** [comment]
We have an application on top of menu which collects rejected calls in a list. The user may use it to call back.

**Pull request merged**